domains/transfer-agreements/TransferAgreements.API/API/ApiModels/Requests/CreateTransferAgreementValidator.cs

```diff
@@ -1,36 +1,35 @@
 using System;
 using FluentValidation;
 
-namespace API.ApiModels.Requests
+namespace API.ApiModels.Requests;
+
+public class CreateTransferAgreementValidator : AbstractValidator<CreateTransferAgreement>
 {
-    public class CreateTransferAgreementValidator : AbstractValidator<CreateTransferAgreement>
+    public CreateTransferAgreementValidator(string senderTin)
```

Comment:
You can use DI to get the HttpContext, see https://stackoverflow.com/questions/56387086/net-core-api-is-it-possible-to-conditionally-validate-a-model-depending-on-the for an example.
Also see here how Fluent Validator validators are added to DI in ASP.NET Core: https://docs.fluentvalidation.net/en/latest/aspnet.html

This will mean that the controller can take an instance of `IValidator<CreateTransferAgreement>` using DI and not know the actual implementation `IValidator<Person>`.

Reply:
I have Chosen to Dependency inject the ihttpcontext as suggested. I will let you review the solution, and close the issue, if you are satisfied with the outcome.